internal/traffic/lb.go

```diff
@@ -12,7 +12,7 @@ type LoadBalancer interface {
 	OpenSocket(id string) (Socket, error)
 	// SendSocket sends a reader to an open socket. If the
 	// socket is not open it returns ErrNotFound.
-	SendSocket(ctx context.Context, socketID string, r io.ReadCloser) error
+	SendSocket(ctx context.Context, socketID string, r io.Reader) error
```

Comment:
sorry about this. i released that SendSocket needs to block until the reader is completely read. which means OpenSocket needs a ReadCloser, but SendSocket only needs an Reader. internally our impl of the SendSocket method can use waitingReadCloser to block.